stackstate_checks_base/stackstate_checks/base/stubs/topology.py

```diff
@@ -37,11 +37,24 @@ def _ensure_instance(self, check_id, instance_key):
             self._snapshots[check_id] = snapshot(instance_key)
         return self._snapshots[check_id]
 
+    def _is_duplicate_relation(self, check_id, instance_key, source_id, target_id, type, data):
+        if relation(source_id, target_id, type, data) in self._ensure_instance(check_id, instance_key)["relations"]:
```

Comment:
same as writing

`return relation(source_id, target_id, type, data) in self._ensure_instance(check_id, instance_key)["relations"]`